CMakeLists.txt

```diff
@@ -13,6 +13,9 @@ endmacro()
 # determine whether to create a debug or release build
 sfml_set_option(CMAKE_BUILD_TYPE Release STRING "Choose the type of build (Debug or Release)")
 
+# Suppress Cygwin legacy warning
+set(CMAKE_LEGACY_CYGWIN_WIN32 0)
```

Comment:
Legacy behavior defined `WIN32` for Cygwin builds, the warning notifies you about that (it's  removed in 2.8.4).